src/DefaultBuilder/test/Microsoft.AspNetCore.TrimmingTests/X509Utilities.cs

```diff
@@ -0,0 +1,40 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Diagnostics.CodeAnalysis;
+using System.Linq;
+using System.Reflection;
+using Microsoft.AspNetCore.Builder;
+
+#nullable enable
+
+public static class X509Utilities
+{
+    public static bool HasCertificateType
+    {
+        get
+        {
+            var certificateType = GetType("System.Security.Cryptography", "System.Security.Cryptography.X509Certificates.X509Certificate");
+
+            // We're checking for members, rather than just the presence of the type,
+            // because Debugger Display types may reference it without actually
+            // causing a meaningful binary size increase.
+            return certificateType is not null && GetMembers(certificateType).Any();
```

Comment:
Another approach we could take here is to disable the feature switch `System.Diagnostics.Debugger.IsSupported` (`DisabledFeatureSwitches` in the .proj file) and ensure this type is fully trimmed. This will get us closer to what AOT does, because that feature switch is disabled by default for PublishAot.

https://github.com/dotnet/runtime/blob/ce689d9189d549e249f4bb5e3d9aeef322464bfb/src/coreclr/nativeaot/BuildIntegration/Microsoft.NETCore.Native.targets#L256-L257

Reply:
I thought about that but worried it would hide problems.  I could add a separate test for that flag, if you think it's worthwhile.

Reply:
> but worried it would hide problems

What kind of problems?

This mode would more closely align with the Native AOT behavior. So it seems the more preferrable to me. The fact that we are rooting types for Debugging published apps is less desirable to verify (IMO).